### PR TITLE
Fixed Incorrect Quiz Question in Binary Search Module

### LINKED ISSUE #3446: Misleading Binary Search Quiz Question
https://usaco.guide/silver/binary-search?lang=cpp#quiz

Screen Shot 2023-02-18 at 6 54 33 PM

Depending on the implementation, it could be $\Omega(\log n)$ in the best case. Probably best to include a code snippet as part of the question.

### PR DESCRIPTION
**Pull request by dustin-miao**
Closes #3446 

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [ ] I have tested my code.
- [ ] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [ ] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

File: content/3_Silver/Binary_Search.mdx [hunk 1]
```diff
@@ -1140,7 +1140,61 @@ for _ in range(query_num):
...
+```
+</CPPSection>
+<JavaSection>
+```java
+boolean binary_search(int x, int[] a) {
+	int l = 0, r = a.length - 1;
+	while (l < r) {
+		int m = (l + r) / 2;
+		if (a[m] == x)
+			return true;
```
Comment (SansPapyrus683) [inline content/3_Silver/Binary_Search.mdx:1170]: braces

File: content/3_Silver/Binary_Search.mdx [hunk 2]
```diff
@@ -1140,7 +1140,61 @@ for _ in range(query_num):
...
+			r = m - 1;
+	}
+	return false;
+}
+```
+</CPPSection>
+<JavaSection>
+```java
+boolean binary_search(int x, int[] a) {
+	int l = 0, r = a.length - 1;
```
Comment (SansPapyrus683) [inline content/3_Silver/Binary_Search.mdx:1166]: separate lines

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: content/3_Silver/Binary_Search.mdx [hunk 3]
```diff
@@ -1149,12 +1149,13 @@ bool binary_search(int x, std::vector<int> &a) {
 	int l = 0, r = a.size() - 1;
```
Comment (SansPapyrus683) [inline content/3_Silver/Binary_Search.mdx:1149]: separate here

Comment (SansPapyrus683) [reply content/3_Silver/Binary_Search.mdx:1149]: & for the python asw

**Review by bqi343 — APPROVED** [review]

**Pull request merged**